bagitobjecttransfer/recordtransfer/views.py

```diff
@@ -609,22 +610,30 @@ def _accept_contents(file_upload):
             accepted is False. The 'clamav' key is a dict itself that has a
             'reason' and a 'status' key.
     '''
-    # TODO: Implement with clamd, like so:
-    '''
-    scan_results = clamd_socket.instream(file_upload)
-    status, reason = scan_results['stream']
-    if (status != 'OK'):
+    clamd_socket = clamd.ClamdNetworkSocket(CLAMD_HOST, CLAMD_PORT)
+    try:
+        scan_results = clamd_socket.instream(file_upload.file)
+        status, reason = scan_results['stream']
+        if status != 'OK':
+            return {
+                'accepted': False,
+                'error': 'Malware found in file',
+                'verboseError': gettext(
+                    'The file "{0}" was identified to contain malware! This issue '
+                    'will be sent to the administrator'.format(file_upload)
+                ),
+                'clamav': {
+                    'reason': reason,
+                    'status': status,
+                }
+            }
+    except (clamd.BufferTooLongError, clamd.ConnectionError) as exc:
+        LOGGER.error("Unable to scan file (%s)", uploadfiles, exc_info=exc)
         return {
             'accepted': False,
-            'error': 'Malware found in file',
+            'error': 'Unable to scan file for malware',
             'verboseError': gettext(
-                'The file "{0}" was identified to contain malware! This issue '
-                'will be sent to the administrator'
-            ),
-            'clamav': {
-                'reason': reason
-                'status': status
-            }
+                'Unable to scan the file "{0}" due to exception {1}'.format(file_upload, exc)
```

Comment:
I don't think we should expose the exception contents to the user. If this occurs in other places we should address those

Reply:
Fair enough, so I'm logging this error already. Do you want to start having a way to send a message to administrators? Also, do we accept the files if Clamav stops working?

Reply:
You should have accession to the `UploadSession` for the transfer. You can call `remove_session_uploads()` on that session to remove all the files that may have been uploaded to the server